src/ripple/rpc/handlers/WalletPropose.cpp

```diff
@@ -31,9 +31,30 @@
 #include <ripple/rpc/handlers/WalletPropose.h>
 #include <ed25519-donna/ed25519.h>
 #include <boost/optional.hpp>
+#include <cmath>
+#include <map>
 
 namespace ripple {
 
+double
+shannon_entropy (std::string const& input)
```

Comment:
If we look at cases where each letter has an equal probability, this calculates:

S*C*lg(S/C), where S=string length and C is the count.

(1) If use unique letters, C=1 and we get S*lg(S).
(2) If we use two letters, C=S/2, and we get S^2*lg(S)/2.

Intuitively, I expect case (1) to have larger entropy than case (2). But I need about 20 characters to get 80 bits from case (1) and only about 13 to get 80 bits from case (2).

Is that right?

Allows 13 alternating "ababa.." to pass this test seems like too low a bar.

(Hopefully I didn't make a silly math mistake that invalidates above. Even so, we should still calculate the shortest allowable pass work with this definition of entropy.)